rust/src/api.rs

```diff
@@ -230,6 +236,7 @@ mod tests {
         pub value: Option<Vec<u8>>,
     }
 
+    #[cfg(feature = "std")]
```

Comment:
Okay, so these tests are not run in no_std mode.

It gets the code to pass (and I guess I can merge it), but it would be nice to run the whole test suite with no-std

Reply:
I think it is necessary to test here in the std environment, for `File` ,`io` and some other libraries are not available in `core` and `alloc`, but only in the standard library.
Relate tourtial: https://docs.rust-embedded.org/book/intro/no-std.html

Reply:
Ah... you are very right.

I guess none of these work in no-std but it would be nice to have equivalent tests.
Maybe you can use [`include_bytes`](https://doc.rust-lang.org/std/macro.include_bytes.html), and then get rid of all the `load_file` stuff as well (and have tests work in both environments)